python/ray/data/tests/test_transform_pyarrow.py

```diff
@@ -485,16 +485,17 @@ def test_fallback_to_pandas_on_incompatible_data(
 
 
 @pytest.mark.parametrize(
-    "op, data",
+    "op, data, should_fail",
     [
-        ("map", [1, 2**100]),
-        ("map_batches", [[1.0], [2**4]]),
+        ("map", [1, 2**100], False),
+        ("map_batches", [[1.0], [2**4]], True),
```

Comment:
Why does this still fail, even with the object extension type?

Reply:
Great question!

So this test is kinda weird in a sense that it is crafted such that UDF produces multiple blocks with different schemas and when we're trying to combine these blocks, Arrow complains

Reply:
Updated the test adding more cases/comments